File: c/questao13.c
#include <stdio.h>
#include <math.h>

double calcular_raiz_quadrada(double n, double precisao) {
    double b = 2.0;
    double p;

    while (1) {
        p = (b + n / b) / 2.0;
        if (fabs(p * p - n) < precisao)
            return p;
        b = p;
    }
}

int main() {
    double numero;
    double precisao = 0.0001;

    printf("Digite um número para calcular a raiz quadrada: ");
    scanf("%lf", &numero);

    if (numero < 0) {
        printf("O número deve ser não-negativo.\n");
        return 1;
    }

    double raiz_aproximada = calcular_raiz_quadrada(numero, precisao);

    printf("A raiz quadrada aproximada de %.2lf é: %.2lf\n", numero, raiz_aproximada);

    return 0;
}

File: python/questao13.py
def calcular_raiz_quadrada(n, precisao=0.0001):
    b = 2.0

    while True:
        p = (b + n / b) / 2.0
        if abs(p * p - n) < precisao:
            return p
        b = p

def main():
    numero = float(input("Digite um número para calcular a raiz quadrada: "))
    precisao = 0.0001

    if numero < 0:
        print("O número deve ser não-negativo.")
        return 1

    raiz_aproximada = calcular_raiz_quadrada(numero, precisao)

    print(f"A raiz quadrada aproximada de {numero:.2f} é: {raiz_aproximada:.2f}")

if __name__ == "__main__":
    main()
